pkg/engine/engine.go

```diff
@@ -354,6 +354,7 @@ func (e *Engine) setDefaults(ctx context.Context) {
 	}
 	e.notifyVerifiedResults = true
 	e.notifyUnverifiedResults = true
+	e.retainFalsePositives = true
```

Comment:
"Why didn't TruffleHog find my secret" is a commonly asked question (e.g., #3579). I think it makes sense to log _everything_ by default. Requiring people to know about and opt-in to a special flag can be confusing.

What do you think?